### PR TITLE
ECCI-420 ECCI-421 ECCI-423 ECCI-424

### PR DESCRIPTION
**Pull request by Polynya**
## Include a summary of what this merge request involves (*)
Allow pages to be set as restricted content so that anonymous users only see the title, images and a link to sign in. Teasers in components also show a sign-in link.

- All content types have a new boolean field localgov_restricted_content
- If the flag is not true then the pages ancestors are checked e.g. if a newsroom is restricted then all its news articles are too
- All content types have two 'fields' in display modes to display a message and/or sign-in link
- Sign-in links need styling

A config split for DDEV has been added so that Devel can be enabled.

### Why have you taken the approach?
New code is in a module called localgov_restricted_content so that it will be easier to move to LGD later.

### Any particular problem areas?
I haven't restricted blocks that are displayed on node pages. Some of these will have to be hidden.
## If necessary, please include any relevant screenshots (If not already available on the JIRA ticket).
image

# This PR has been testing in the following browsers
- [ ] Edge
- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] IE 11 (Windows)
- [ ] iOS Chrome
- [ ] iOS Safari
- [ ] Android Chrome
- [ ] Android Firefox
- [ ] Android default

### CODE REVIEW

File: modules/custom/localgov_restricted_content/localgov_restricted_content.info.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: "Essex Restricted Content"
```
Comment (siliconmeadow) [inline modules/custom/localgov_restricted_content/localgov_restricted_content.info.yml:1]: Do I remember correctly that the module is likely to end up as an LGD module? If so, should the `description` and `package` name reflect that now?

File: modules/custom/localgov_restricted_content/localgov_restricted_content.module [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+function localgov_restricted_content_theme($existing, $type, $theme, $path) {
+  return [
+    'localgov_restricted_content_sign_in' => [
+      'render element' => 'elements',
+      ],
+    'localgov_restricted_content_sign_in_with_message' => [
+      'render element' => 'elements',
+    ],
+  ];
+}
```
Comment (siliconmeadow) [inline modules/custom/localgov_restricted_content/localgov_restricted_content.module:112]: Out of curiosity, does Drupal's coding standards no longer require a trailing new line?

File: modules/custom/localgov_restricted_content/templates/localgov-restricted-content-sign-in-with-message.html.twig [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<!--cludooff: index-->
```
Comment (siliconmeadow) [inline modules/custom/localgov_restricted_content/templates/localgov-restricted-content-sign-in-with-message.html.twig:1]: `cludooff/cludoon` would be Essex specific. So if it becomes an LGD module, I suppose there would need to be a theme override?

File: modules/custom/localgov_restricted_content/templates/localgov-restricted-content-sign-in.html.twig [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<!--cludooff: index-->
```
Comment (siliconmeadow) [inline modules/custom/localgov_restricted_content/templates/localgov-restricted-content-sign-in.html.twig:1]: As previous comment re: `cludoon/cludooff`

**Review by siliconmeadow — COMMENTED** [review]
No blockers - just some comments.

File: modules/custom/localgov_restricted_content/localgov_restricted_content.module (hunk 2 above)
Comment (Polynya) [reply modules/custom/localgov_restricted_content/localgov_restricted_content.module]: Codesniffer didn't spot it, don't know why.

**Pull request merged**